include/linker/devicetree_regions.h

```diff
@@ -50,6 +57,23 @@
 	ORIGIN = DT_REG_ADDR(node_id),	      \
 	LENGTH = DT_REG_SIZE(node_id)
 
+/**
+ * @brief Declare a memory section from the device tree nodes with
+ *	  compatible 'zephyr,memory-region'
+ *
+ * @param node_id devicetree node identifier
+ */
+#define _SECTION_DECLARE(node_id)								\
+	_DT_SECTION_NAME(node_id) DT_REG_ADDR(node_id) (NOLOAD) :				\
+	{											\
+		_DT_SECTION_START(node_id) = .;							\
+		KEEP(*(_DT_SECTION_NAME(node_id)))						\
+		KEEP(*(_DT_SECTION_NAME(node_id).*))						\
+		_DT_SECTION_END(node_id) = .;							\
+	}											\
```

Comment:
Can we add a `> LINKER_DT_NODE_REGION_NAME(node_id)` here so we get linker errors if the section size exceeds the memory region size?

Reply:
yes but I actually removed that so that the generated sections are the same when using `CONFIG_CMAKE_LINKER_GENERATOR=y` and when that is not used.

Reply:
ok, fixed and I aligned the CMake generated files as well.